### PR TITLE
Add prepare script

### PR DESCRIPTION
**Pull request by shekharkhedekar**

### CODE REVIEW

**Comment by shekharkhedekar** [comment]
@Logerfo this attempts to fix https://github.com/Logerfo/newline-action/issues/57.  The [docs](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#adding-actions-toolkit-packages) mention that a `package-lock.json` should be included.

File: .gitignore [hunk 1]
```diff
@@ -60,6 +60,4 @@ typings/
 .env
 
 # next.js build output
-.next
-
-package-lock.json
+.next
```
Comment (Logerfo) [inline .gitignore:63]: This is funny. I guess I should recursively add the newline-action to the newline-action repo itself.

**Comment by shekharkhedekar** [comment]
Still trying to get this to work without any luck...seems to be specific to `js-yaml`

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,2680 @@
...
+      "dev": true,
+      "dependencies": {
+        "sprintf-js": "~1.0.2"
+      }
+    },
+    "node_modules/@eslint/eslintrc/node_modules/js-yaml": {
+      "version": "3.14.1",
+      "resolved": "https://registry.npmjs.org/js-yaml/-/js-yaml-3.14.1.tgz",
+      "integrity": "sha512-okMH7OXXJ7YrN9Ok3/SXrnu4iX9yOk+25nqX4imS2npuvTYDmo/QEZoqwZkYaIDk3jVvBOTOIEgEhaLOynBS9g==",
+      "dev": true,
```
Comment (Logerfo) [inline package-lock.json:180]: Maybe this isn't supposed to be here, but I don't know why it is. Inspired by https://github.com/Polymer/polymer-modulizer/pull/437/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519L556.

**Comment by shekharkhedekar** [comment]
I stand corrected - if I reorder imports, it can't find `@actions/core`. Something is happening that is failing `npm install`. Digging more.

**Comment by shekharkhedekar** [comment]
@Logerfo  Had to do a couple of things to get it to work:
- Add a `prepare` npm script [similar to the example action](https://github.com/actions/javascript-action/blob/main/package.json#L8)
- Update `@actions/github` to the latest, and use `github.getOctokit`

Tested on [this PR](https://github.com/mrengy/vanguard-history/pull/76). 

Please take a look. Thanks!

**Comment by Logerfo** [comment]
@shekharkhedekar do you mind sanitizing the PR so the diff can be minimum? Seems like changes to main.js are not needed.
Also, the commit on your test PR seems empty. The bot is not supposed to push empty changes. Is that correct?

**Comment by shekharkhedekar** [comment]
> @shekharkhedekar do you mind sanitizing the PR so the diff can be minimum? Seems like changes to main.js are not needed. Also, the commit on your test PR seems empty. The bot is not supposed to push empty changes. Is that correct?

@Logerfo cleaned up `main.js` - there was one change. As far as the empty commit, I see that as well - not sure why that's happening.

**Comment by Logerfo** [comment]
Thank you for your contribution!
